### PR TITLE
fix: DEV-3298: Repeater performance improvement

### PR DESCRIPTION
**Pull request by juliosgarbi**

### CODE REVIEW

**Review by bmartel — CHANGES_REQUESTED** [review]
Looks good, some cleanup to do on Pagination component.

File: src/core/settings/keymap.json [hunk 1]
```diff
@@ -154,5 +154,13 @@
   "video:hop-forward": {
     "key": "shift+alt+right",
     "description": "Hop Forward"
+  },
+  "repeater:next-page": {
+    "key": "alt+right",
+    "description": "Hop Forward"
```
Comment (bmartel) [inline src/core/settings/keymap.json:160]:
```suggestion
    "description": "Next Page"
```

File: src/core/settings/keymap.json [hunk 2]
```diff
@@ -154,5 +154,13 @@
...
+  },
+  "repeater:previous-page": {
+    "key": "alt+left",
+    "description": "Hop Forward"
```
Comment (bmartel) [inline src/core/settings/keymap.json:164]:
```suggestion
    "description": "Previous Page"
```

**Review by bmartel — CHANGES_REQUESTED** [review]
Just a couple changes to the descriptions of the hotkeys to match the usage. Rest is ready to go.

**Review by bmartel — APPROVED** [review]

**Review by bmartel — APPROVED** [review]
Works well, nice one.

File: src/tags/visual/Repeater.js [hunk 3]
```diff
@@ -46,5 +46,6 @@
  * @meta_description Customize Label Studio with the Repeater tag to repeat similar data blocks to accelerate labeling for machine learning and data science projects.
  * @param {string} on                  - Data field object with array with similar data
  * @param {string} [indexFlag={{idx}}] - Placeholder for array index in params of underlying tags
+ * @param {default|pagination} [mode]  - Select display mode between default and paginated
```
Comment (hlomzik) [inline src/tags/visual/Repeater.js:49]: I'd name it "list" maybe?

File: src/tags/object/PagedView.js [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+const DEFAULT_PAGE_SIZE = 1;
+
+const HtxPagedView = observer(({ item }) => {
+  const [page, setPage] = useState(1);
+  const [pageSize, setPageSize] = useState(DEFAULT_PAGE_SIZE);
+
+  useEffect(() => {
+    setPageSize(getStoredPageSize('repeater', DEFAULT_PAGE_SIZE));
+
+    setTimeout(() => {
```
[the remark this answers is not in the document]
Comment (hlomzik) [reply src/tags/object/PagedView.js:97]: but it's still here, in a next tick. why?

**Review by hlomzik — CHANGES_REQUESTED** [review]

File: src/core/settings/keymap.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (juliosgarbi) [reply src/core/settings/keymap.json]: This file is outdated

File: src/tags/visual/Repeater.js (hunk 3 above)
Comment (juliosgarbi) [reply src/tags/visual/Repeater.js:49]: Done

**Pull request merged**